src/simulation/m_riemann_solvers.fpp

```diff
@@ -2108,6 +2149,14 @@ contains
                                     end do
                                 end if
 
+                                if (low_Mach == 2) then
+                                    zcoef = min(1d0, max(vel_L_rms**0.5d0/c_L, vel_R_rms**0.5d0/c_R))
+                                    qL_prim_rs${XYZ}$_vf(j, k, l, contxe + idx1) = 5d-1*((vel_L(idx1) + vel_R(idx1)) + zcoef*(vel_L(idx1) - vel_R(idx1)))
+                                    qR_prim_rs${XYZ}$_vf(j + 1, k, l, contxe + idx1) = 5d-1*((vel_L(idx1) + vel_R(idx1)) + zcoef*(vel_R(idx1) - vel_L(idx1)))
```

Comment:
these `if(low_mach = 1` (or `2`) routines seem to be repeated for all of the types of Riemann solver. they should be turned into their own functions or subroutines.